lib/kintsugi/apply_change_to_project.rb

```diff
@@ -115,17 +115,35 @@ def add_missing_component_if_valid(parent_component, change_name, change, change
 
     def replace_component_with_new_type(parent_component, name_in_parent_component, change)
       old_component = parent_component.send(name_in_parent_component)
-
-      new_component = parent_component.project.new(
-        Module.const_get("Xcodeproj::Project::#{change["isa"][:added]}")
-      )
+      new_component = component_of_new_type(parent_component, change, old_component)
 
       copy_attributes_to_new_component(old_component, new_component)
 
       parent_component.send("#{name_in_parent_component}=", new_component)
       new_component
     end
 
+    def component_of_new_type(parent_component, change, old_component)
+      if change["isa"][:added] == "PBXFileReference"
+        path = change["path"] || old_component.path
```

Comment:
should be `(change["path"]  && change["path"][:added]) || old_component.path`, because if the path changed the new path will be in `:added`, like with `isa`.